platform/view/services/db/driver/sql/base.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+	"runtime/debug"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type base struct {
+	db         *sql.DB
+	txn        *sql.Tx
+	txnLock    sync.Mutex
+	debugStack []byte
+	table      string
+}
+
+func (db *base) Close() error {
+	logger.Info("closing database")
+
+	// TODO: what to do with db.txn if it's not nil?
+
+	err := db.db.Close()
+	if err != nil {
+		return errors.Wrap(err, "could not close DB")
```

Comment:
`github.com/pkg/errors` has been archived and we should transition to move away from it. 
Simple wrapping can be done with the `fmt` package. For example: `fmt.Errorf("decompress %v: %w", name, err)`